runtime/src/invoke_dpe.rs

```diff
@@ -71,11 +71,14 @@ impl InvokeDpeCmd {
                     }
                     cmd.execute(dpe, &mut env, locality)
                 }
-                Command::DeriveChild(cmd) => {
-                    Drivers::is_dpe_context_threshold_exceeded(
-                        pl0_pauser, flags, locality, dpe, false,
-                    )?;
-                    if DeriveChildCmd::changes_locality(&cmd)
+                Command::DeriveContext(cmd) => {
+                    // DeriveContext can only create new contexts if the recursive flag is not set.
```

Comment:
nit: I think this comment is a bit confusing. Maybe something like:
```suggestion
                    // If the recursive flag is not set, DeriveContext will generate a new context.
                    // If recursive _is_ set, it will extend the existing one, which will not count
                    // against the context threshold.
```

Reply:
Done